execution-engine/src/hcall/common.rs

```diff
@@ -275,578 +160,57 @@ impl<T> From<std::sync::PoisonError<T>> for HostProvisioningError {
-
-    /// Returns the data source frame (containing raw data, source and package
-    /// IDs) associated with the Nth registered data source as stored in the
-    /// host provisioning state, expressed by `index`.  Returns `None` iff
-    /// `index` is not the index of any registered data source.
-    pub(crate) fn get_current_data_source(&self, index: usize) -> Option<&DataSourceMetadata> {
-        if index > self.get_current_data_source_count() {
-            return None;
-        }
-        Some(&self.data_sources[index])
+    pub fn new(
+        vfs : Arc<Mutex<VFS>>,
+    ) -> Self {
+        Self { vfs }
     }
 
-    /// Returns the data source frame (containing raw data, source and package
-    /// IDs) associated with the Nth registered data source as stored in the
-    /// host provisioning state, expressed by `index`.  Returns `None` iff
-    /// `index` is not the index of any registered data source.
-    pub(crate) fn get_current_stream_source(&self, index: usize) -> Option<&DataSourceMetadata> {
-        if index > self.get_current_stream_source_count() {
-            return None;
-        }
-        Some(&self.stream_sources[index])
-    }
-
-    /// Returns the data source frame (containing raw data, source and package
-    /// IDs) associated with the Nth registered data source as stored in the
-    /// host provisioning state, expressed by `index`.  Returns `None` iff
-    /// `index` is not the index of any registered data source.
-    pub(crate) fn get_previous_result(&self) -> Option<&Vec<u8>> {
-        self.previous_result.as_ref()
-    }
-
-    /// Returns the client IDs of clients who are able to request platform
-    /// shutdown, per the global policy.
-    #[inline]
-    pub(crate) fn get_expected_shutdown_sources(&self) -> &Vec<u64> {
-        self.expected_shutdown_sources.borrow()
-    }
-
-    /// Returns `true` iff a program module has been registered by the
-    /// provisioning process.
-    #[inline]
-    pub(crate) fn is_program_registered(&self) -> bool {
-        match &self.program_module {
-            None => false,
-            Some(_module) => true,
-        }
-    }
-
-    /// Returns `true` iff a program's memory is registered by the provisioning
-    /// process.
-    #[inline]
-    pub(crate) fn is_memory_registered(&self) -> bool {
-        match &self.memory {
-            None => false,
-            Some(_memory) => true,
-        }
-    }
-
-    /// Returns `true` iff a result has been written by the executing program.
-    #[inline]
-    pub(crate) fn is_result_registered(&self) -> bool {
-        self.result != None
-    }
-
-    #[inline]
-    pub(crate) fn is_able_to_shutdown(&self) -> bool {
-        self.expected_shutdown_sources.is_empty()
-    }
-
-    /// Returns an optional result computed by the WASM program when it has
-    /// finished executing.
-    #[inline]
-    pub(crate) fn get_result(&self) -> Option<&Vec<u8>> {
-        self.result.as_ref()
+    /// Append to a file.
```

Comment:
Is this really appending to a file?  I think the comment looks suspiciously like it was copied-over from `append_file_base`.

Reply:
I have modified this part of rustdoc. also add mark to indicate they are glue code and will be removed in wasi change.